specification/src/main/asciidoc/platform/Interoperability.adoc

```diff
@@ -2,7 +2,7 @@
 == Interoperability
 
 This chapter describes the interoperability
-requirements for the Jakarta™ EE Platform.
+features for the Jakarta™ EE Platform.
```

Comment:
At first this appeared to me as noise, but after a few moments I think it's clear you're attempting to eliminate the blanket statement "everything in this chapter is a requirement", so you could dial things in more specifically within the chapter.  Did I get that right?

Reply:
Yes, that's what I was trying to do...

Reply:
Are there any parts of (now pruned/removed) `interoperability` that we need to specify for Jakarta EE 9?  

Compatible implementations may support RMI-IIOP + Corba but might not implement CSIv2, so application developers may get confused by seeing a chapter on `interoperability` in the EE 9 Platform spec.

I will have more feedback than I added below if we do keep this chapter.

Reply:
There is another section in the Platform Spec that specifically addresses EJB 4.0 requirements.  The removal of the Distributed Interop was mentioned in that section.

This Interop chapter is wonky.  It brings in aspects of several types of interop -- network protocols, data formats, and the corba stuff.  We're trying to soften the Corba/OMG stuff and leave the other aspects of interop as-is.  Again, not perfect, but not sure what else to do at this point.  We're kind of in a limbo state...

Reply:
@kwsutter If we could avoid using the single `Interoperability` word as much, as it is overloaded:

1. Replace title with `Application Interoperability` or `App Interop`.
1. Replace `the Jakarta Enterprise Beans interoperability protocol based on IIOP` with `Corba`.
1. Replace `the Jakarta Enterprise Beans interoperability protocol in the Jakarta Enterprise Beans specification` with `Corba`.

We still would use `interoperability` in the generic sense for elsewhere in this document.

Reply:
The addition of the "application interoperability" I think has good intent, but not the results we're hoping for and our best bet is to leave it as-is.

 - We haven't said how "application interoperability" is the same as or different than "interoperability" which is still heavily used in this section and elsewhere.  As someone who just went on a 22-year dig through an IIOP, RMI, RMI/IIOP, RMI-IIOP, CORBA/IIOP, CORBA-IIOP, CORBA, RMI-JMRP deciphering journey I have heightened empathy for trying to reverse engineer intent behind all the variations.  Sometimes they are synonyms, sometimes antonyms.
 - There were phases in EJB where terms were renamed and not carried out consistently, leaving the old term hanging around, no explanation if it was the same or different and therefore making the topic less clear, not more clear.  @Emily-Jiang points this out on one other title, but even changing that one title does not eliminate the heavy reliance on the bare word "interoperability."  Unless we change them all, we are just adding more "interoperability" confusion, not reducing confusion.
 - The word "interoperability" is used in the chapter near the nouns client, server, protocol, legacy system, and application.  Many of us would consider selecting "application" as the primary description to be odd.  In other areas we deliberately attempt to describe client, server and application as three distinct things,

We avoid the least confusion by just leaving it as is.  If we do want to rename it, we'd need to all agree it's the right new term, do it consistently throughout the spec so the old term is left hanging undefined or undifferentiated, and explain the motivation for the change in the changelog.

I don't think any of this is worth it.

Reply:
So there's something encouraging, huge +1 to remove "Jakarta Enterprise Beans interoperability protocol based on IIOP" for the reasons above.  We definitely do *not* need another undefined term to describe remote EJB capabilities.

Reply:
Since we cleared up some of the IIOP items, then changing this chapter back to just "Interoperability" is fine with me.  I'll revert that change.